airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -618,16 +618,16 @@ def _graceful_decommission_timeout_object(self) -> Optional[Dict]:
         match = re.match(r"^(\d+)([smdh])$", self.graceful_decommission_timeout)
         if match:
             if match.group(2) == "s":
-                timeout = int(match.group(1))
+                timeout = float(match.group(1))
```

Comment:
The regex is `\d+` so `int()` feels right here

```suggestion
                timeout = int(match.group(1))
```

Reply:
This one I think makes sense as float since it is only passed as a param to something that is a float  
(to hooks/dataproc.py -> update_cluster which takes timeout as an Optional[float] )